ee/desktop/runner/runner.go

```diff
@@ -212,40 +231,99 @@ func (r *DesktopUsersProcessesRunner) Interrupt(interruptError error) {
 	}
 }
 
+// Update handles control server updates for the desktop subsystem
+// This includes data related to the runner itself, and the user processes
 func (r *DesktopUsersProcessesRunner) Update(data io.Reader) error {
-	var desktopStatus client.DesktopUserStatus
+	var desktopStatus DesktopUserStatus
 	if err := json.NewDecoder(data).Decode(&desktopStatus); err != nil {
 		return fmt.Errorf("failed to decode desktop user control data: %w", err)
 	}
 
-	if desktopStatus.Status == "" {
-		return fmt.Errorf("empty desktop status")
+	// Regardless, we will write the menu data out to a file that can be grabbed by
+	// any desktop user processes, either when they refresh, or when they are spawned.
+	if err := r.writeMenuFile(desktopStatus.Menu); err != nil {
+		return err
+	}
+
+	if desktopStatus.Enabled && r.processSpawningEnabled {
```

Comment:
`processSpawningEnabled` controls whether or not desktop user processes are created. The process creation doesn't happen until the execute loop ticker fires again - that's why it's a member field of the runner, so it can be persisted until the execute loop fires. I felt like it was simpler to rely on the existing process creation flow rather than interjecting special case process creation in this spot.

However, I am inconsistent and I am killing the desktop processes here rather than waiting for the ticker to fire; I could move the `killDesktopProcesses` call to `runConsoleUserDesktop` to make process creation/killing follow similar flows, where it is done asynchronously according to the execute loop frequency. That would also simply `toggleDesktop` to essentially be a `processSpawningEnabled` setter method only.

Reply:
> I felt like it was simpler to rely on the existing process creation flow rather than interjecting special case process creation in this spot.

That's a good point!

Though I'm still not sure why you need it in this if. If the status is enabled, you can iterate `r.uidProcs`. It should be empty if `r.processSpawningEnabled`. Though maybe that array doesn't actually clean up.

Reply:
`r.uidProcs` now gets cleared when the runner is interrupted, or told to kill all the user processes (like when desktop is being disabled). This got simpler now with the desktop flags separated.